src/store.rs

```diff
@@ -8,130 +9,159 @@ use zkevm_opcode_defs::{
 
 use crate::utils::address_into_u256;
 
-/// Trait for storage operations inside the VM, this will handle the sload and sstore opcodes.
-/// This storage will handle the storage of a contract and the storage of the called contract.
-pub trait Storage: Debug {
-    fn decommit(&self, hash: U256) -> Result<Option<Vec<U256>>, StorageError>;
-    fn add_contract(&mut self, hash: U256, code: Vec<U256>) -> Result<(), StorageError>;
+#[derive(Debug, Clone)]
+pub struct L2ToL1Log {
+    pub key: U256,
+    pub value: U256,
+    pub is_service: bool,
+    pub address: H160,
+    pub shard_id: u8,
+    pub tx_number: u16,
+}
+
+pub trait InitialStorage: Debug {
     fn storage_read(&self, key: StorageKey) -> Result<Option<U256>, StorageError>;
-    fn storage_write(&mut self, key: StorageKey, value: U256) -> Result<(), StorageError>;
-    fn storage_drop(&mut self, key: StorageKey) -> Result<(), StorageError>;
-    fn get_state_storage(&self) -> &HashMap<StorageKey, U256>;
-    fn get_all_keys(&self) -> Vec<StorageKey>;
-    fn fake_clone(&self) -> InMemory;
-    fn rollback(&mut self, previous: &dyn Storage) {
-        let keys = previous.get_all_keys();
-        for key in keys {
-            let value = previous.storage_read(key).unwrap();
-            if let Some(value) = value {
-                self.storage_write(key, value).unwrap();
-            }
-        }
-        let current_keys = self.get_all_keys();
-        for key in current_keys {
-            let res = previous.storage_read(key);
-            if let Ok(None) = res {
-                self.storage_drop(key).unwrap();
-            };
-        }
-    }
 }
 
-/// Error type for storage operations.
-#[derive(Error, Debug)]
-pub enum StorageError {
-    #[error("Key not present in storage")]
-    KeyNotPresent,
-    #[error("Error writing to storage")]
-    WriteError,
-    #[error("Error reading from storage")]
-    ReadError,
+#[derive(Debug, Clone)]
+pub struct InitialStorageMemory {
```

Comment:
Let's add some comments  here on what this trait represents and why would you implement it.